### PR TITLE
feat: added logic for GVCF rows

### PR DESCRIPTION
**Pull request by JeremyWesthead**
Added extra logic to override rows which have an arbitrary depth to be null calls. Can be used to enforce null calls on GVCF rows

`gnomonicus` can then use a merged VCF (including GVCF rows) to call null calls. This would avoid FASTA adjudication while retaining the null calls and adding the coverage data. The exact threshold for a null can then be passed as an argument to account for ONT/clockwork differences in `MIN_DP`

### CODE REVIEW

**Review by philipwfowler — COMMENTED** [review]
I can see min_dp is now provided as a parameter to VariantFile; is there a unit test which covers the new functionality? e.g. has the threshold above and then below?

**Comment by JeremyWesthead** [comment]
I just checked and this doesn't quite work with the sundial GVCF rows

File: gumpy/variantfile.py [hunk 1]
```diff
@@ -96,13 +99,23 @@ def __init__(self, record: pysam.libcbcf.VariantRecord, sample: str | int):
...
+                    self.call2 = self.call1
                 self.is_reference = (
-                    True if call1 == 0 and (call1 == call2 or call2 == -1) else False
+                    True
+                    if self.call1 == 0
+                    and (self.call1 == self.call2 or self.call2 == -1)
+                    else False
                 )
-                self.is_heterozygous = True if call1 != call2 else False
+                self.is_heterozygous = True if self.call1 != self.call2 else False
```
[the remark this answers is not in the document]
Comment (JeremyWesthead) [reply gumpy/variantfile.py:118]: That would require a VCF to have a row of `0/.` which I'm not convinced would make logical sense. But yes, if that was the case, we'd get that situation (and this is not actually a change to the logic)

**Review by mcolpus — APPROVED** [review]
Logic looks good to me. Not fully sure if it needs more tests though? Maybe a vcf which uses AD instead of COV, or one triggering the plody 1 gvcf edge case
